src/momento/incubating/aio/simple_cache_client.py

```diff
@@ -38,65 +41,125 @@ async def dictionary_set(
         self,
         cache_name: str,
         dictionary_name: str,
-        dictionary: Dictionary,
-    ) -> CacheDictionarySetResponse:
+        key: Optional[DictionaryKey] = None,
+        value: Optional[DictionaryValue] = None,
+        dictionary: Optional[Dictionary] = None,
+        ttl_seconds: Optional[int] = None,
+        *,
+        refresh_ttl: bool,
+    ) -> Union[CacheDictionarySetUnaryResponse, CacheDictionarySetMultiResponse]:
         """Store dictionary items (key-value pairs) in the cache.
 
         Inserts items from `dictionary` into a dictionary `dictionary_name`.
         Updates (overwrites) values if the key already exists.
 
+        Items may be set in one of two ways:
+        - the function may be run with either a single item to set, `key` and `value, or
+        - multiple items with the `dictionary` argument which accepts a `Dictionary`.
+
+        To illustrate:
+        >>> client.dictionary_set(cache_name, dictionary_name, key="key" value="value")
+        >>> client.dictionary_set(cache_name, dictionary_name, dictionary={"key1": "value1", "key2": "value2"})
+
         Args:
             cache_name (str): Name of the cache to store the dictionary in.
             dictionary_name (str): The name of the dictionary in the cache.
-            dictionary (Dictionary): The items (key-value pairs) to be stored.
+            key (Optional[DictionaryKey], optional): The key to set (unary set). Defaults to None.
+                value (Optional[DictionaryValue], optional): The value to set (unary set). Defaults to None.
+            dictionary (Optional[Dictionary], optional): The items (key-value pairs) to be stored (multi set).
+                Defaults to None.
+            ttl_seconds (Optional[int], optional): Time to live in seconds for the dictionary
+                as a whole.
+            refresh_ttl (bool): If, when performing an update, to refresh the ttl.
 
         Returns:
-            CacheDictionarySetResponse: data stored in the cache
+            Union[CacheDictionarySetUnaryResponse, CacheDictionarySetMultiResponse]: data stored in the cache
```

Comment:
Also by using 2 named methods you would be able to return simple return types instead of dynamic unions